modules/plugins/utility/motion/precognition/precognition.nix

```
@@ -0,0 +1,78 @@
+{lib, ...}: let
+  inherit (lib.options) mkEnableOption mkOption literalExpression;
+  inherit (lib.types) attrsOf listOf str bool int submodule;
+in {
+  options.vim.utility.motion.precognition = rec {
+    enable = mkEnableOption "precognition.nvim plugin";
+
+    startVisible = mkOption {
+      type = bool;
+      description = "Whether to start 'precognition' automatically.";
+      default = true;
+    };
+
+    showBlankVirtLine = mkOption {
+      type = bool;
+      description = "Whether to show a blank virtual line when no movements are shown.";
+      default = true;
+    };
+
+    highlightColor = mkOption {
+      type = attrsOf str;
+
+      example = literalExpression ''
+        { link = "Comment"; }
+        # or
+        { foreground = "#0000FF"; background = "#000000"; };
+      '';
+      default = {link = "Comment";};
+      description = ''
+        The highlight for the virtual text.
+
+        To find more about this option, please see
+        [the documentation](https://github.com/tris203/precognition.nvim).
```

```suggestion
        [the plugin documentation](https://github.com/tris203/precognition.nvim).
```

"the documentation" implies nvf documentation, better to clarify here